bigtable/tests/unit/test_batcher.py

```diff
@@ -0,0 +1,250 @@
+# limitations under the License.
+
+
+import unittest
+
+import mock
+
+from ._testing import _make_credentials
+
+
+class TestMutationsBatcher(unittest.TestCase):
+    from grpc import StatusCode
+
+    PROJECT_ID = 'project-id'
+    INSTANCE_ID = 'instance-id'
+    INSTANCE_NAME = ('projects/' + PROJECT_ID + '/instances/' + INSTANCE_ID)
+    TABLE_ID = 'table-id'
+    TABLE_NAME = INSTANCE_NAME + '/tables/' + TABLE_ID
+
+    # RPC Status Codes
+    SUCCESS = StatusCode.OK.value[0]
+    RETRYABLE_1 = StatusCode.DEADLINE_EXCEEDED.value[0]
+    RETRYABLE_2 = StatusCode.ABORTED.value[0]
+    NON_RETRYABLE = StatusCode.CANCELLED.value[0]
+
+    def _make_responses(self, codes):
+        import six
+        from google.cloud.bigtable_v2.proto.bigtable_pb2 import (
+            MutateRowsResponse)
+        from google.rpc.status_pb2 import Status
+
+        entries = [MutateRowsResponse.Entry(
+            index=i, status=Status(code=codes[i]))
+            for i in six.moves.xrange(len(codes))]
+        return MutateRowsResponse(entries=entries)
+
+    @staticmethod
+    def _get_target_class():
+        from google.cloud.bigtable.table import Table
+
+        return Table
+
+    def _make_table(self, *args, **kwargs):
+        return self._get_target_class()(*args, **kwargs)
+
+    @staticmethod
+    def _get_target_client_class():
+        from google.cloud.bigtable.client import Client
+
+        return Client
+
+    def _make_client(self, *args, **kwargs):
+        return self._get_target_client_class()(*args, **kwargs)
+
+    def test_constructor(self):
+        from google.cloud.bigtable.batcher import MutationsBatcher
+
+        credentials = _make_credentials()
+        client = self._make_client(project='project-id',
+                                   credentials=credentials, admin=True)
+
+        instance = client.instance(instance_id=self.INSTANCE_ID)
+        table = self._make_table(self.TABLE_ID, instance)
+
+        mutation_batcher = MutationsBatcher(table)
+        self.assertEqual(table.table_id, mutation_batcher.table.table_id)
+
+    def test_add_row(self):
+        from google.cloud.bigtable.batcher import MutationsBatcher
+        from google.cloud.bigtable.row import DirectRow
+        from google.cloud.bigtable_admin_v2.gapic import (
+            bigtable_table_admin_client)
+        from google.cloud.bigtable_v2.gapic import (
+            bigtable_client)
+
+        data_api = bigtable_client.BigtableClient(mock.Mock())
+        table_api = mock.create_autospec(
+            bigtable_table_admin_client.BigtableTableAdminClient)
+        credentials = _make_credentials()
+        client = self._make_client(project='project-id',
+                                   credentials=credentials, admin=True)
+        client._table_data_client = data_api
+        client._table_admin_client = table_api
+        instance = client.instance(instance_id=self.INSTANCE_ID)
+        table = self._make_table(self.TABLE_ID, instance)
```

Comment:
This test can be simplified by mocking a Table object.  There's no need to mock all of the other things in this and other tests.